### PR TITLE
Make PacketType the size of u8

### PR DESCRIPTION
**Pull request by coolreader18**
I noticed that methods on ConnectPacket weren't properly setting flags values (`set_dup(false)` wouldn't clear the bit), and I also realized it should be possible to pack it into one byte instead of 2.

### CODE REVIEW

File: src/control/packet_type.rs [hunk 1]
```diff
@@ -55,98 +55,145 @@ pub enum ControlType {
...
+    }
+}
+
 impl PacketType {
-    /// Creates a packet type
+    /// Creates a packet type. Returns None if `flags` is an invalid value for the given
+    /// ControlType as defined by the [MQTT spec].
+    ///
+    /// [MQTT spec]: http://docs.oasis-open.org/mqtt/mqtt/v3.1.1/os/mqtt-v3.1.1-os.html#_Table_2.2_-
+    pub fn new(t: ControlType, flags: u8) -> Option<PacketType> {
```
Comment (zonyitoo) [inline src/control/packet_type.rs:90]: It shouldn't be `Option`. You may want to use `Result` instead.

**Comment by zonyitoo** [comment]
@coolreader18 What do you think? `new` should return `Result`, right?

It has been a while since last release.

**Comment by coolreader18** [comment]
Well I originally had it returning `Result<, PacketTypeError>`, but I realized that it only had one error condition, and given that you passed the flags you probably didn't need access to them in the error as well. However I could just have a unit error type, or one that just wraps the flags.

**Comment by zonyitoo** [comment]
Although it only is an unit error type, but it should be an `Err` in this case.

A wrapper better for implementing `Debug` for this error type, I think.

**Pull request merged**